src/btree/bt_cursor.c

```diff
@@ -512,12 +512,48 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     return (0);
 }
 
+/*
+ * __wt_btcur_evict_reposition --
+ *     Try to evict the page and reposition the cursor on the saved key.
+ */
+int
+__wt_btcur_evict_reposition(WT_CURSOR_BTREE *cbt)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    cursor = &cbt->iface;
+    session = CUR2S(cbt);
+
+    /*
+     * Try to evict the page and then reposition the cursor back to the page for operations with
+     * snapshot isolation and for pages that require urgent eviction. Snapshot isolation level
+     * maintains a snapshot allowing the cursor to point at the correct value after a reposition
+     * unlike read committed isolation level.
+     */
+    if (session->txn->isolation == WT_ISO_SNAPSHOT && F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      __wt_page_evict_soon_check(session, cbt->ref, NULL)) {
+        WT_RET(__wt_cursor_localkey(cursor));
+        WT_RET(__cursor_reset(cbt));
+        ret = __wt_btcur_search(cbt, false);
+        /*
+         * If we got a WT_ROLLBACK it may be because there is a lot of cache pressure and the
```

Comment:
may be is a bit vague.

Reply:
Updated in [54b348d](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/54b348d4c92fec2b365cc62cb895dfcfc219b494)